src/utils/fastmath.h

```diff
@@ -68,4 +68,18 @@ inline float FastLog(const float a) {
 // Fast approximate exp(x). Does only limited range checking.
 inline float FastExp(const float a) { return FastPow2(1.442695040f * a); }
 
+inline float FastInvSqrt(const float number)
+{ 
+  const float x2 = number * 0.5F;
+  const float threehalfs = 1.5F;
+
+  union {
+    float f;
+    uint32_t i;
+  } conv = {number}; // member 'f' set to value of 'number'.
+  conv.i  = 0x5f3759df - ( conv.i >> 1 );
+  conv.f  *= ( threehalfs - ( x2 * conv.f * conv.f ) );
+  return conv.f;
+}
+
```

Comment:
The following variation avoids undefined behavior and gives identical code with gcc:
```
inline float FastInvSqrt(const float number)
{ 
  const float x2 = number * 0.5F;
  const float threehalfs = 1.5F;
  float f;
  uint32_t i;
  memcpy(&i, &number, sizeof(float));
  i  = 0x5f3759df - ( i >> 1 );
  memcpy(&f, &i, sizeof(float));
  return f * ( threehalfs - ( x2 * f * f ) );
}
```
Also the `_mm_rsqrt_ss()` alternative should be provided here - this should be faster, maybe the version tested earlier was doing some unnecessary float to double conversions?

Reply:
`return numerator * GetP() * FastInvSqrt(x * x * x);` was tested and 3% slower.

Reply:
@borg323 what would be the proper way to use `_mm_rsqrt_ss()` to avoid conversions?

Reply:
The code you suggested seems fine:
```
float FastInvSqrt(const float number)
{ 
    return _mm_cvtss_f32(_mm_rsqrt_ss(_mm_set_ss(number)));
}
```
Similarly the following variation generates identical code:
```
float FastInvSqrt(const float number)
{ 
    return _mm_cvtss_f32(_mm_rsqrt_ss(_mm_load_ss(&number)));
}
```
With gcc, the code is:
```
        pxor    xmm1, xmm1
        movss   xmm1, xmm0
        movaps  xmm0, xmm1
        rsqrtss xmm0, xmm1
        ret
```
Both clang and icc do a better job
```
        rsqrtss xmm0, xmm0
        ret
```
I expect when inlined the gcc code would also be reduced to the minimum.

Reply:
Trying `return _mm_cvtss_f32(_mm_rsqrt_ss(_mm_load_ss(&number)));` using gcc is 2.5% slower than the current implementation.

Reply:
Using clang-8 it is even slower (9% as compared to current implementation in gcc).